### PR TITLE
Added support for before and after hunts

### PR DESCRIPTION
**Pull request by adamgoerdt**

### CODE REVIEW

File: hunting-buddy.lic [hunk 1]
```diff
@@ -49,16 +49,32 @@ class HuntingBuddy
       end
 
       args = info['args']
+      before_actions = info['before']
+      after_actions = info['after']
       duration = info[:duration]
       stop_on_skills = info['stop_on']
 
+      #Execute scripts to run before the hunt
```
Comment (rpherbig) [inline hunting-buddy.lic]: I recommend extracting this into a helper method. That keeps this main method more about flow, abstracting the details out to the helper.

File: hunting-buddy.lic [hunk 2]
```diff
@@ -49,16 +49,32 @@ class HuntingBuddy
...
+      before_actions.each do |before_action|
+        action_parts = before_action.split(" ")
+        script_name = action_parts.shift
+        wait_for_script_to_complete(script_name, action_parts)
+      end
+
       next if all_skills_at_cap?(stop_on_skills)
```
Comment (rpherbig) [inline hunting-buddy.lic]: This line (and line 65) are two cases in which after actions would not occur. Is that ok, or will that cause problems?

**Comment by rpherbig** [comment]
Do you have any YAML files which show an example of how this is used that you would be willing to upload?

File: hunting-buddy.lic (hunk 2 above)
Comment (adamgoerdt) [inline hunting-buddy.lic]: It might be me not understanding the flow, but wouldn't release_cyclics also not be called if the stop_on_skills are hit?

**Comment by adamgoerdt** [comment]
Here are a couple examples from my yamls: http://pastebin.com/UfHhhWcv

File: hunting-buddy.lic (hunk 1 above)
Comment (adamgoerdt) [inline hunting-buddy.lic]: Agreed.  Having a hard time thinking of what to name the function, though.

Comment (rpherbig) [inline hunting-buddy.lic]: `execute_actions`?

File: hunting-buddy.lic (hunk 2 above)
Comment (rpherbig) [inline hunting-buddy.lic]: You are correct

**Pull request merged**